### PR TITLE
Made Pesto logo fade smoothly.

### LINKED ISSUE #5684: Gallery, Pesto: pesto icon in app bar jumps/janks on resize
"On dynamic resize of toolbar, would be great if PESTO label faded during the resize animation, so that the icon could simply shrink rather than shrink but then (jump cut) expand"

### PR DESCRIPTION
**Pull request by dragostis**
Added a fading effect to the Peste logo from the Gallery.
Fixes #5684.

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/pesto_demo.dart [hunk 1]
```diff
@@ -123,17 +123,38 @@ class _RecipeGridPageState extends State<RecipeGridPage> {
         builder: (BuildContext context, BoxConstraints constraints) {
           final Size size = constraints.biggest;
           final double appBarHeight = size.height - statusBarHeight;
-          final String logo = appBarHeight >= 70.0 ? _kMediumLogoImage : _kSmallLogoImage;
           // Extra padding. Calculated to give about 16px on the bottom for the
           // `small` logo at its native size, and 30px for the `medium`.
           final double extraPadding = min(0.19 * appBarHeight + 5.4, 40.0);
+          final double t = (appBarHeight - 56.0) / (_kAppBarHeight - 56.0);
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/pesto_demo.dart]: Use kToolbarHeight

File: examples/flutter_gallery/lib/demo/pesto_demo.dart [hunk 2]
```diff
@@ -123,17 +123,38 @@ class _RecipeGridPageState extends State<RecipeGridPage> {
...
+          const double kOverlap = 0.28;
+          final Interval interval = new Interval(0.5 - kOverlap, 0.5 + kOverlap);
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/pesto_demo.dart]: const Interval

**Comment by HansMuller** [comment]
LGTM

File: examples/flutter_gallery/lib/demo/pesto_demo.dart [hunk 3]
```diff
@@ -123,17 +127,38 @@ class _RecipeGridPageState extends State<RecipeGridPage> {
...
+          const double kOverlap = 0.28;
+          const Interval interval = const Interval(0.5 - kOverlap, 0.5 + kOverlap);
+
           return new Padding(
             padding: new EdgeInsets.only(
               top: statusBarHeight + 0.5 * extraPadding,
               bottom: extraPadding
             ),
             child: new Center(
-              child: new Image.asset(logo, fit: ImageFit.scaleDown)
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/pesto_demo.dart]: don't we want to only fade out the text? this will render two images on top of each other, no?

**Comment by dragostis** [comment]
We should have an implementation that splits the logo in image + label and continue from there. Closing.

**Pull request closed without merging**